src/micro_ros_arduino.h

```diff
@@ -102,10 +102,12 @@ struct micro_ros_agent_locator {
 };
 
 static inline void set_microros_wifi_transports(char * ssid, char * pass, char * agent_ip, uint agent_port){
-	while (WiFi.begin(ssid, pass) != WL_CONNECTED)
-	{
-    	delay(500);
-	}
+
+	WiFi.begin(ssid, pass);
+
+    while (WiFi.status() != WL_CONNECTED) {
```

Comment:
WHy aren't we trying in a loop to connect using `begin` instead of trying once and checking the status?

Reply:
`begin` configures the network `ssid` and `pass` and checks the connection using the `status`, as you can see here: https://github.com/arduino-libraries/WiFi/blob/master/src/WiFi.cpp#L98

There is no need to reconfigure this values again, when we can just check the status of the connection.

Reply:
Nice, thanks for the clarification